pandas/core/frame.py

```diff
@@ -3503,78 +3503,111 @@ def dropna(self, axis=0, how='any', thresh=None, subset=None,
+            * 1, or 'columns' : Drop columns which contain missing value.
+
             Pass tuple or list to drop on multiple axes.
         how : {'any', 'all'}, default 'any'
-            Determine if label is removed when we have at least one NA or all
-            NA.
-            * any : If any NA values are present, drop that label.
-            * all : If all values are NA, drop that label.
-        thresh : int, default None
+            Determine if row or column is removed from DataFrame, when we have
+            at least one NA or all NA.
+
+            * 'any' : If any NA values are present, drop that row or column.
+            * 'all' : If all values are NA, drop that row or column.
+        thresh : int, optional
             Require that many non-NA values.
-        subset : array-like
+        subset : array-like, optional
             Labels along other axis to consider, e.g. if you are dropping rows
             these would be a list of columns to include.
-        inplace : boolean, default False
+        inplace : bool, default False
             If True, do operation inplace and return None.
 
         Returns
         -------
         DataFrame
             DataFrame with NA entries dropped from it.
 
-        See also
+        See Also
         --------
-        isna: show missing values
-        notna : show existing (non-missing) values
+        DataFrame.isna: show missing values
+        DataFrame.notna : show existing (non-missing) values
+        isna : top-level isna
+        notna : top-level notna
 
         Examples
         --------
-        >>> df = pd.DataFrame([[np.nan, 2, np.nan, 0], [3, 4, np.nan, 1],
-        ...                    [np.nan, np.nan, np.nan, 5]],
-        ...                   columns=list('ABCD'))
+        Drop the rows where at least one element is missing.
+
+        >>> df = pd.DataFrame([[1., 2.], [3., np.nan]], columns=list('AB'))
         >>> df
-             A    B   C  D
-        0  NaN  2.0 NaN  0
-        1  3.0  4.0 NaN  1
-        2  NaN  NaN NaN  5
+             A    B
+        0  1.0  2.0
+        1  3.0  NaN
+        >>> df.dropna()
+             A    B
+        0  1.0  2.0
 
-        Drop the columns where all elements are nan:
+        Drop the columns where at least one element is missing.
 
-        >>> df.dropna(axis=1, how='all')
-             A    B  D
-        0  NaN  2.0  0
-        1  3.0  4.0  1
-        2  NaN  NaN  5
+        >>> df.dropna(axis=1)
+             A
+        0  1.0
+        1  3.0
 
-        Drop the columns where any of the elements is nan
+        Drop the rows where all elements are missing.
 
-        >>> df.dropna(axis=1, how='any')
-           D
-        0  0
-        1  1
-        2  5
+        >>> df = pd.DataFrame([[1., 2.], [np.nan, np.nan], [5., np.nan]],
+        ...                   columns=list('AB'))
+        >>> df
+             A    B
+        0  1.0  2.0
+        1  NaN  NaN
+        2  5.0  NaN
```

Comment:
Personally, I'd define this dataframe at the beginning (before the first explanations), and reuse it for all the examples. Or one with 3 columns, if you think it's not descriptive enough when using subset.

I think reusing the same data will make much more obvious the differences about every case.

Reply:
Agreed, will rework the examples. Was uncertain, whether to write simpler & specific ones for each parameter or just a general one, that might be more complex at the first glance but can be reused. So will take your proposed approach.

Reply:
Yes, I think this is a good idea